src/SIL.Harmony/DataModel.cs

```diff
@@ -147,8 +150,18 @@ ValueTask<bool> ISyncable.ShouldSync()
     private async Task UpdateSnapshots(Commit oldestAddedCommit, Commit[] newCommits)
     {
         await _crdtRepository.DeleteStaleSnapshots(oldestAddedCommit);
-        var modelSnapshot = await GetProjectSnapshot(true);
-        var snapshotWorker = new SnapshotWorker(modelSnapshot.Snapshots, _crdtRepository);
+        Dictionary<Guid, Guid?> snapshotLookup;
+        //this is a performance optimization to avoid loading all the snapshots, this number is somewhat arbitrary
+        if (newCommits.Length > 10)
```

Comment:
I feel like it would be pretty easy to pull the entity IDs out of the new commits and use those to query for relevant snapshots. What do you think?

Reply:
That's such a good idea I also had it, and then forgot about it 😅. I do want to avoid that when there's too many commits, but I suppose that would be better than querying all snapshots, so maybe I'll just do that here always.

Reply:
Alright, first I tried out just always querying the ids, but that caused my perf tests to fail (this is why you write those), now those are specifically testing for performance of one change at a time (the most common case), so I left the code to only query when there's more than 10 commits, but did the filtering like you suggested

Reply:
I was expecting you to pass the entity IDs deeper into the SQL.
I.e. It might be a performance boost to already filter `LatestSnapshots`.